### PR TITLE
build: add iOS support

### PR DESCRIPTION
**Pull request by mbouron**
While it does not support the iOS simulator yet, it has the advantage to enable the MoltenVK backend and text.

Depends https://github.com/NopeForge/nope.gl/pull/256.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -706,13 +824,34 @@ def _glslang_setup(cfg):
...
+                f"-DCMAKE_INSTALL_PREFIX={cfg.prefix}",
+                f"-DCMAKE_OSX_SYSROOT=iphoneos",
+                f"-DCMAKE_OSX_ARCHITECTURES={cfg.ios_abi}",
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-DBUILD_EXTERNAL=OFF",
+                "-DENABLE_OPT=OFF",
+                "-S",
+                cfg.externals["glslang"],
+                "-B",
+                _get_builddir(cfg, "glslang"),
```
Comment (ubitux) [inline configure.py:871]: can we factor that out?

File: configure.py [hunk 2]
```diff
@@ -831,6 +979,10 @@ def _nopegl_setup(cfg):
         extra_library_dirs += [op.join(cfg.prefix, "lib")]
         extra_include_dirs += [op.join(cfg.prefix, "include")]
 
+    if cfg.host == "iOS":
```
Comment (ubitux) [inline configure.py:1010]: `elif`

File: configure.py [hunk 3]
```diff
@@ -105,6 +106,60 @@ def _gen_android_cross_file(cfg):
         fp.write(cross_file)
 
 
+_IOS_ABI_MAP = dict(
+    aarch64="arm64",
+    x86_64="x86_64",
+)
+
+
+_IOS_CROSS_FILE_TPL = textwrap.dedent(
```
Comment (ubitux) [inline configure.py:115]: Make sure the existing ini files are dropped

File: configure.py [hunk 4]
```diff
@@ -482,6 +550,24 @@ def _boringssl_setup(cfg):
...
+                "-GNinja",
+                f"-DCMAKE_BUILD_TYPE={build_type}",
+                f"-DCMAKE_INSTALL_PREFIX={cfg.prefix}",
+                f"-DCMAKE_OSX_SYSROOT=iphoneos",
+                f"-DCMAKE_OSX_ARCHITECTURES={cfg.ios_abi}",
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-S",
+                cfg.externals["boringssl"],
+                "-B",
+                _get_builddir(cfg, "boringssl"),
```
Comment (ubitux) [inline configure.py:574]: can you check if there isn't an SSL builtin on iOS?

File: configure.py [hunk 5]
```diff
@@ -496,148 +582,165 @@ def _boringssl_install(cfg):
...
+            "--target-os=android",
+            f"--cross-prefix={cfg.android_ndk_bin}{op.sep}llvm-",
+            f"--cc={cfg.android_ndk_bin}{os.sep}{cfg.android_compiler}-clang",
+        ]
+    elif cfg.host == "iOS":
+        extra_cflags += f"-arch {cfg.ios_abi} -mios-version-min={cfg.ios_version}"
+        extra_ldflags += f"-arch {cfg.ios_abi} -mios-version-min={cfg.ios_version}"
+        extra_args += [
+            "--enable-videotoolbox",
+            "--enable-videotoolbox",
```
Comment (ubitux) [inline configure.py:711]: dup

File: configure.py [hunk 6]
```diff
@@ -1231,7 +1402,10 @@ def get_env(self):
         env = {}
         env["PATH"] = sep.join((self.venv_bin_path, "$(PATH)"))
         env["PKG_CONFIG_PATH"] = self.pkg_config_path
-        if self.host == "Android":
+        if self.host in ["Android"]:
```
Comment (ubitux) [inline configure.py:1433]: please choose between `in` and `==` ;)

File: ngl-tools/ngl-desktop.c [hunk 7]
```diff
@@ -468,59 +454,18 @@ static struct ngl_scene *get_default_scene(const char *host, const char *port)
...
-    s->addr = rp;
-
-    if (listen(s->sock_fd, 0) < 0) {
-        perror("listen");
+    s->socket = SDLNet_TCP_Open(&ip);
+    if(!s->socket) {
+        fprintf(stderr, "SDLNet_TCP_Open: %s\n", SDLNet_GetError());
         return NGL_ERROR_IO;
     }
+    fprintf(stderr, "coucou");
```
Comment (ubitux) [inline ngl-tools/ngl-desktop.c:468]: coucou

File: configure.py (hunk 3 above)
Comment (mbouron) [reply configure.py:115]: Done.

File: configure.py [hunk 8]
```diff
@@ -105,6 +106,60 @@ def _gen_android_cross_file(cfg):
...
+    objcpp_args = ['-arch', '{abi}', '-isysroot', sysroot]
+    objcpp_link_args = ['-arch', '{abi}', '-isysroot', sysroot]
+    [properties]
+    root = root
+    [binaries]
+    c = 'clang'
+    cpp = 'clang++'
+    objc = 'clang'
+    strip = 'strip'
+    pkgconfig = 'pkg-config'
```
[the remark this answers is not in the document]
Comment (mbouron) [reply configure.py:136]: Done.

File: configure.py (hunk 6 above)
Comment (mbouron) [reply configure.py:1433]: Done.

File: configure.py (hunk 2 above)
Comment (mbouron) [reply configure.py:1010]: Fixed.

File: configure.py (hunk 1 above)
Comment (mbouron) [reply configure.py]: Done.

File: ngl-tools/ngl-desktop.c (hunk 7 above)
Comment (mbouron) [reply ngl-tools/ngl-desktop.c]: Removed :D

File: configure.py (hunk 1 above)
Comment (mbouron) [reply configure.py]: Also did the same for boringssl.

File: configure.py (hunk 4 above)
Comment (mbouron) [reply configure.py]: Done by using `--enable-securetransport`.

Comment (mbouron) [reply configure.py]: Nevermind, while it compiled, the API used by this tls backend are deprecated by Apple. Thus, I re-enabled the build  of the boringssl lib.

File: configure.py (hunk 5 above)
Comment (mbouron) [reply configure.py]: Fixed.

**Review by ubitux — APPROVED** [review]

**Pull request merged**